util/testutil/integration/run.go

```diff
@@ -0,0 +1,60 @@
+package integration
+
+import (
+	"os/exec"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Sandbox interface {
+	Address() string
+	PrintLogs(*testing.T)
+	Cmd(...string) *exec.Cmd
+}
+
+type Worker interface {
+	New() (Sandbox, func() error, error)
+	Name() string
+}
+
+type Test func(*testing.T, Sandbox)
+
+var defaultWorkers []Worker
+
+func register(w Worker) {
+	defaultWorkers = append(defaultWorkers, w)
+}
+
+func List() []Worker {
+	return defaultWorkers
+}
+
+func Run(t *testing.T, testCases map[string]Test) {
```

Comment:
I think we can use `[]Test` instead of `map[string]Test`.
The function name can be fetched by using reflection: https://stackoverflow.com/questions/7052693/how-to-get-the-name-of-a-function-in-go

Reply:
Added